.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: ci
+
+on: [pull_request]
+
+jobs:
+
+  pull:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+    - name: "Check file encoding"
+      run:
+        git remote add upstream https://github.com/pingcap/blog.git;
+        git fetch upstream;
+        wget https://raw.githubusercontent.com/CharLotteiu/pingcap-docs-checks/main/check-file-encoding.py;
+        python3 check-file-encoding.py $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md')
+    - name: Markdown lint
+      uses: avto-dev/markdown-lint@v1
+      with:
+        config: './.github/markdownlint.yml'
+        args: '.'
+    - name: "Check control characters"
+      run:
+        wget https://raw.githubusercontent.com/CharLotteiu/pingcap-docs-checks/main/check-control-char.py;
+        python3 check-control-char.py $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md')
+    - name: "Check unclosed tags"
+      run:
+        wget https://raw.githubusercontent.com/CharLotteiu/pingcap-docs-checks/main/check-tags.py;
+        echo "checked files are $(git diff-tree --name-only --no-commit-id -r upstream/master...HEAD -- '.md')";
+        python3 check-tags.py $(git diff-tree --name-only --no-commit-id -r upstream/master...HEAD -- '.md')
+    - name: "Check manual line breaks"
+      run:
+        wget https://raw.githubusercontent.com/CharLotteiu/pingcap-docs-checks/main/check-manual-line-breaks.py;
+        python3 check-manual-line-breaks.py $(git diff-tree --name-only --no-commit-id -r upstream/master..HEAD -- '*.md')
+
+  verify-links:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12'
+    - name: Verify internal links
+      run: ./hack/verify-links-doc.sh
+    # the script reports "#top" as dead anchor.
+    # - name: Verify internal link anchors
```

Comment:
Do you need to verify `internal link anchors`, if yes, uncomment this step, otherwise, remove file `hack/verify-link-anchors.sh`.

Reply:
Yes. We need to verify `internal link anchors`. 

But currently, we have lots of `#top` anchors in old blog articles (e.g. https://pingcap.com/blog/2017-07-11-tidbinternal1#top), which are not regular markdown anchors and are thus reported "dead anchor" by this script.

I will find a workaround to address this issue. But before that, I had to comment out this step.

Reply:
Got it. The #top is a useless anchor, I think you can remove it redirectly.